src/webapi.ts

```diff
@@ -21,6 +21,80 @@ export interface WebApiResponse<Result> {
   severity: string | undefined;
 }
 
+export function updateQuery(query: CmkQuery): void {
+  // convert legacy query (context/params) to new requestSpec interface
+  if (query.context !== undefined && query.params !== undefined) {
+    // we need to replace the object in place, in order to remove the context
+    // and params properties, otherwise both requestSpec and context/params are defined.
+    query.requestSpec = requestSpecFromLegacy(query.context, query.params);
+    delete query.context;
+    delete query.params;
+  }
+}
+
+function transform_negated(
+  context_property: Record<string, string | undefined> | undefined,
+  key_name: string
+): NegatableOption | undefined {
+  if (context_property === undefined) {
+    return undefined;
+  }
+  return {
+    value: context_property[key_name] || '',
+    negated: context_property[`neg_${key_name}`] === 'on',
+  };
+}
+
+export function requestSpecFromLegacy(context: Context, params: Params): RequestSpec {
+  const rs: RequestSpec = {};
+  if (context.host !== undefined) {
+    rs.host_name = context.host.host;
+  }
+
+  if (context.siteopt !== undefined) {
+    rs.site = context.siteopt.site;
+  }
+
+  rs.host_name_regex = transform_negated(context.hostregex, 'host_regex');
+
+  if (context.service !== undefined) {
+    rs.service = context.service.service;
+  }
+
+  rs.service_regex = transform_negated(context.serviceregex, 'service_regex');
+
+  rs.host_in_group = transform_negated(context.opthostgroup, 'opthost_group');
+  rs.service_in_group = transform_negated(context.optservicegroup, 'optservice_group');
+
+  if (context.host_labels !== undefined) {
+    const host_labels = JSON.parse(context.host_labels.host_label);
```

Comment:
Something in the back of my head tells me that this field was Python notation, not JSON.

But since we are only supporting queries we generated, we don't need to worry about that case, don't we? Old values are written using [`JSON.stringify`](https://github.com/tribe29/grafana-checkmk-datasource/blob/v2.0.3/src/components/filters.tsx#L139)

Reply:
it's also json in checkmk: https://github.com/tribe29/checkmk/blob/master/cmk/gui/utils/labels.py#L47-L67 
you can create an invalid value here: https://github.com/tribe29/checkmk/blob/master/tests/unit/cmk/gui/plugins/visuals/test_filters.py#L311 and see the backtrace there.